modules/publicize/publicize.php

```diff
@@ -245,33 +407,146 @@ function show_options_popup( $service_name, $connection ) {
 		return false;
 	}
 
-	function user_id() {
-		global $current_user;
-		return $current_user->ID;
+	/**
+	 * Whether the Connection is "valid" wrt Facebook's requirements.
+	 *
+	 * Must be connected to a Page (not a Profile).
+	 * (Also returns true if we're in the middle of the connection process)
+	 *
+	 * @param object|array The Connection object (WordPress.com) or array (Jetpack)
+	 * @return bool
+	 */
+	function is_valid_facebook_connection( $connection ) {
+		if ( $this->is_connecting_connection( $connection ) ) {
+			return true;
+		}
+		$connection_data = $connection->get_meta( 'connection_data' );
```

Comment:
Will this work in the Jetpack case (`$connection` being an array there)?

Reply:
No - it definitely won't :) Thanks!

Reply:
Fixed in 5087018f6d0670ffd6c043bc96902e4bbdffeb62.